pkg/reconciler/trigger/controller.go

```diff
@@ -73,26 +73,28 @@ func NewConstructor(dataresidencyss *dataresidency.StoreSingleton) Constructor {
 }
 
 func newController(ctx context.Context, cmw configmap.Watcher, drs *dataresidency.Store) *controller.Impl {
+	// Parse all env vars of interest
 	var env envConfig
 	if err := envconfig.Process("", &env); err != nil {
 		logging.FromContext(ctx).Fatal("Failed to process env var", zap.Error(err))
 	}
 
 	triggerInformer := triggerinformer.Get(ctx)
 
+	var client *pubsub.Client = nil
```

Comment:
nit: pointer zero value is `nil`, so `var client *pubsub.Client = nil` is redundant, you could just say `var client *pubsub.Client`

Reply:
I tried to avoid implicit default value, as it is confusing to me. Pointer and object is OK, `int` and `bool` can be quite confusing. But if this is a Go style, I can change that. Or will `client := *pubsub.Client(nil)` works?

Reply:
Just looking through the code base, the convention seems to be `var name *type`. I can't find a single use of `var name *type = nil` nor `name := *type(nil)`.

Reply:
This is not a big deal though, it's up to you

Reply:
Fixed as suggested, thanks